autogpt_platform/frontend/src/app/auth/integrations/oauth_callback/route.ts

```diff
@@ -19,13 +20,15 @@ export async function GET(request: Request) {
           message: `Incomplete query: ${searchParams.toString()}`,
         };
 
+  console.debug("Sending message to opener:", message);
+
   // Return a response with the message as JSON and a script to close the window
   return new NextResponse(
     `
     <html>
       <body>
         <script>
-          window.postMessage(${JSON.stringify(message)});
+          window.opener.postMessage(${JSON.stringify(message)}, "*");
```

Comment:
probably not needed but was suggested

Reply:
>[!NOTE]
Always provide a specific targetOrigin, not *, if you know where the other window's document should be located. Failing to provide a specific target could disclose data to a malicious site.

Reply:
>[!NOTE]
A target origin can be specified using the [targetOrigin](https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-windowpostmessageoptions-targetorigin) member of options. If not provided, it defaults to "/". This default restricts the message to same-origin targets only.

If the origin of the target window doesn't match the given target origin, the message is discarded, to avoid information leakage. To send the message to the target regardless of origin, set the target origin to "*".